### PR TITLE
Set ellipsis_inclusive_range_patterns lint to warn

### PR DESCRIPTION
**Pull request by memoryruins**
Continuing #54910, the `ellipsis_inclusive_range_patterns` lint is set to warn.

r? @Centril

### CODE REVIEW

File: src/test/run-pass/binding/pat-ranges.rs [hunk 1]
```diff
@@ -8,7 +8,7 @@ mod m {
 }
 
 fn main() {
-    if let NUM1 ... m::NUM2 = 10 {} else { panic!() }
-    if let ::NUM1 ... ::m::NUM2 = 11 {} else { panic!() }
-    if let -13 ... -10 = 12 { panic!() } else {}
```
Comment (Centril) [inline src/test/run-pass/binding/pat-ranges.rs]: Use `#![allow(...)]` here but also extend the tests to exercise `..=` by duplicating all 3 lines above.

File: src/test/run-pass/associated-consts/associated-const-range-match-patterns.rs [hunk 2]
```diff
@@ -12,15 +12,15 @@ impl HasNum for Foo {
...
     assert!(match 0 {
-        -1 ... <Foo as HasNum>::NUM => true,
+        -1 ..= <Foo as HasNum>::NUM => true,
         _ => false,
     });
     assert!(match 1 {
-        <Foo as HasNum>::NUM ... <Foo>::NUM => true,
+        <Foo as HasNum>::NUM ..= <Foo>::NUM => true,
         _ => false,
     });
```
Comment (Centril) [inline src/test/run-pass/associated-consts/associated-const-range-match-patterns.rs]: Same here.

File: src/test/run-pass/inc-range-pat.rs [hunk 3]
```diff
@@ -1,9 +1,9 @@
 // Test old and new syntax for inclusive range patterns.
```
Comment (Centril) [inline src/test/run-pass/inc-range-pat.rs]: See this line ^--- 😃. We want `#![allow(...)]` here.

File: src/test/run-pass/issues/issue-15881-model-lexer-dotdotdot.rs [hunk 4]
```diff
@@ -6,32 +6,32 @@
```
Comment (Centril) [inline src/test/run-pass/issues/issue-15881-model-lexer-dotdotdot.rs]: See line 4, the point of this test is to exercise the lexer wrt. DOTDOTDOT. We want `#![allow(...)]` here.

File: src/test/ui/lint/issue-54538-unused-parens-lint.rs [hunk 5]
```diff
@@ -15,14 +15,14 @@ fn main() {
     }
```
Comment (Centril) [inline src/test/ui/lint/issue-54538-unused-parens-lint.rs]: Use `#![allow(...)]` and duplicate all `match`es here to exercise `...` in the new ones as well as `..=` in the old ones.

**Review by Centril — CHANGES_REQUESTED** [review]

**Comment by Centril** [comment]
@rust-lang/lang N.B. This transitions a Rust 2018 idiom lint towards `Warn` by default instead of `Allow`. The justification was given by @aturon in https://github.com/rust-lang/rust/issues/28237#issuecomment-304145654.

I believe @joshtriplett's comment in https://github.com/rust-lang/rust/pull/61203#issuecomment-496556150 applies here as well:
> Note that under normal circumstances, I'd say we need a crater run to assess the impact of even a new warning, as that *can* cause build failures (`deny(warnings)`). However, as this was announced and advertised as part of the Rust 2018 edition, I think we can reasonably introduce it as an idiom lint, to be enabled as part of that family of lints.

This is probably less widespread than `dyn Trait` also, so we should be good.

File: src/test/run-pass/inc-range-pat.rs (hunk 3 above)
Comment (memoryruins) [reply src/test/run-pass/inc-range-pat.rs]: 🤦

**Review by Centril — APPROVED** [review]
Thanks! r=me rollup with green travis.

**Comment by memoryruins** [comment]
It was worth a shot :)

**Comment by Centril** [comment]
Hehe :)

@bors r+ rollup

**Pull request merged**